### ISSUE TITLE
Giscus not rendering if used with `class="giscus"`

### ISSUE THREAD

**Issue by alifeee** [giscus/giscus#1199]
Sometimes, Giscus does not load. I see nothing in the console to explain why.

To replicate:

- visit <https://blog.alifeee.co.uk/hull-bus-sign/>
- refresh OR ctrl + refresh OR open in incognito and refresh/ctrl+refresh
- you may have to refresh a few times to get it to happen

My Giscus ([source](https://github.com/alifeee/blog/blob/8a216496ad1402c5a97dcf395dfaf4b7248cf847/hull-bus-sign/index.html#L82-L97))

```html
<script
      src="https://giscus.app/client.js"
      data-repo="alifeee/blog"
      data-repo-id="R_kgDOJaqeLQ"
      data-category="comments"
      data-category-id="DIC_kwDOJaqeLc4CZQST"
      data-mapping="og:title"
      data-strict="1"
      data-reactions-enabled="1"
      data-emit-metadata="0"
      data-input-position="top"
      data-theme="light"
      data-lang="en"
      crossorigin="anonymous"
      async
    ></script>
```

My Giscus element ([source](https://github.com/alifeee/blog/blob/8a216496ad1402c5a97dcf395dfaf4b7248cf847/hull-bus-sign/index.html#L132-L138))

```html
<div id="giscus" class="giscus">
        <h2>Comments!</h2>
        <p>
          There should be a comment section here. If there isn't, refresh the
          page (and consider telling me :).
        </p>
      </div>
```

### Environment

Tested on latest Firefox & Chrome, on Windows and mobile. The problem is on all.

### What it *should* look like

![image](https://github.com/giscus/giscus/assets/13833017/6ea3db16-2c3c-461a-b5f6-efed9901d913)

### What it looks like often (when Giscus does not load)

![image](https://github.com/giscus/giscus/assets/13833017/4ec9643f-246c-44ef-98b4-d53b4630c25d)

### Info

The console messages are the same in both cases. I do not know how to further investigate this issue.

**Comment by alifeee** [comment]
(I think) I fixed this with https://github.com/alifeee/blog/commit/fc5c2ade64a0e9b858fec7b30f9f3ddfaee42ad4, by moving the giscus script to where I wanted the comments. It seems to work now.

## Before

```diff
<html>
  <head>
-   <script
-     src="https://giscus.app/client.js"
-     ...
-   ></script>
  </head>
  <body>
    ...
-   <div class="giscus"></div>
+   <script
+     src="https://giscus.app/client.js"
+     ...
+   ></script>
  </body>
</html>
```

I did it the other way originally because of the [docs](https://giscus.app/)

![image](https://github.com/giscus/giscus/assets/13833017/4e498f51-76f0-4eb0-9756-9a8a29592769)

However, it seems there are issues with this. I could see nothing in the console when it was broken.

**Comment by alifeee** [comment]
fixed by fc5c2ade64a0e9b858fec7b30f9f3ddfaee42ad4
